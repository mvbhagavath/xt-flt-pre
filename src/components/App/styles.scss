.search-sort-section {
  display: flex;
  justify-content: space-between;
  margin: 10px 0;
  padding: 15px;
}
