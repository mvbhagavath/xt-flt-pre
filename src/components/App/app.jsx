import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import CharacterList from "../CharacterList";
import Search from "../Search";
import Sort from "../Sort";
import SidePanel from "../SidePanel/SidePanel";
import { getData } from "../../utils/api";
import { fetchDataSuccess, fetchDataFail } from "../../redux/actions/fetchData";
import "./styles.scss";

const App = () => {
  const { data } = useSelector((state) => state.fetchData);
  const characters = data?.results;
  const dispatch = useDispatch();

  useEffect(() => {
    /* istanbul ignore else */
    if (!characters) {
      dispatch(getData("/character", fetchDataSuccess, fetchDataFail));
    }
  });

  return characters && characters.length > 0 ? (
    <div className="app-container container-fluid">
      <div className="row">
        <div className="col-lg-2">
          <SidePanel />
        </div>
        <div className="col-lg-10">
          <section className="row search-sort-section">
            <Search />
            <Sort />
          </section>
          <CharacterList characters={characters} />
        </div>
      </div>
    </div>
  ) : (
    <div>Loading....</div>
  );
};

export default App;
