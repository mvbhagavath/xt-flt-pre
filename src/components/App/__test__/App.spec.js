import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { render, fireEvent, waitFor, screen } from "@testing-library/react";
import App from "..";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  fetchData: {
    setFilters: {
      species:[]
    }
  },
});

describe("App", () => {
  it("should render app without crashing", () => {
    const wrapper = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
