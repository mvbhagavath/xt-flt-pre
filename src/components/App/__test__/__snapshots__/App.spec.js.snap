// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render app without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app container-fluid"
      >
        <header
          class="app__header"
        >
          <img
            alt="rick and morty"
            class="app__logo"
            src="test-file-stub"
          />
          <h1
            class="app__heading"
          >
            Rick and Morty
          </h1>
        </header>
        <div
          class="row"
        >
          <div
            class="col-lg-2 side-panel"
          >
            <div
              class="side-panel__filter-label"
            >
              <span>
                Filters
              </span>
              <button
                class="side-panel__filter-label-icon"
                type="button"
              >
                +
              </button>
            </div>
            <div
              class="side-panel__filter-container 
          "
            >
              <div
                class="filter-group"
              >
                <div
                  class="filter-group__label"
                >
                  Species
                </div>
                <button
                  class="filter-group__filter inactive"
                  type="button"
                >
                  Human
                </button>
                <button
                  class="filter-group__filter inactive"
                  type="button"
                >
                  Animal
                </button>
                <button
                  class="filter-group__filter inactive"
                  type="button"
                >
                  Alien
                </button>
              </div>
              <div
                class="filter-group"
              >
                <div
                  class="filter-group__label"
                >
                  Gender
                </div>
                <button
                  class="filter-group__filter inactive"
                  type="button"
                >
                  Male
                </button>
                <button
                  class="filter-group__filter inactive"
                  type="button"
                >
                  Female
                </button>
                <button
                  class="filter-group__filter inactive"
                  type="button"
                >
                  unknown
                </button>
              </div>
              <div
                class="filter-group"
              >
                <div
                  class="filter-group__label"
                >
                  Origin
                </div>
                <button
                  class="filter-group__filter inactive"
                  type="button"
                >
                  Earth
                </button>
                <button
                  class="filter-group__filter inactive"
                  type="button"
                >
                  Mars
                </button>
                <button
                  class="filter-group__filter inactive"
                  type="button"
                >
                  unknown
                </button>
              </div>
            </div>
          </div>
          <section
            class="col-lg-10 app__main-section"
          >
            <section
              class="row app__search-sort-section"
            >
              <div
                class="search-bar"
              >
                <input
                  class="search-bar__input"
                  id="filled-search"
                  label="Search field"
                  placeholder="Search by name"
                  type="search"
                />
                <button
                  class="search-bar__button"
                  type="button"
                >
                  Search
                </button>
              </div>
              <div
                class="sort"
              >
                <span
                  class="sort__label"
                >
                  Sort By ID
                </span>
                <button
                  class="sort__button "
                  type="button"
                >
                  Asec
                </button>
                <button
                  class="sort__button "
                  type="button"
                >
                  Desc
                </button>
              </div>
            </section>
            <div
              class="app app__message"
            >
              Loading....
            </div>
          </section>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app container-fluid"
    >
      <header
        class="app__header"
      >
        <img
          alt="rick and morty"
          class="app__logo"
          src="test-file-stub"
        />
        <h1
          class="app__heading"
        >
          Rick and Morty
        </h1>
      </header>
      <div
        class="row"
      >
        <div
          class="col-lg-2 side-panel"
        >
          <div
            class="side-panel__filter-label"
          >
            <span>
              Filters
            </span>
            <button
              class="side-panel__filter-label-icon"
              type="button"
            >
              +
            </button>
          </div>
          <div
            class="side-panel__filter-container 
          "
          >
            <div
              class="filter-group"
            >
              <div
                class="filter-group__label"
              >
                Species
              </div>
              <button
                class="filter-group__filter inactive"
                type="button"
              >
                Human
              </button>
              <button
                class="filter-group__filter inactive"
                type="button"
              >
                Animal
              </button>
              <button
                class="filter-group__filter inactive"
                type="button"
              >
                Alien
              </button>
            </div>
            <div
              class="filter-group"
            >
              <div
                class="filter-group__label"
              >
                Gender
              </div>
              <button
                class="filter-group__filter inactive"
                type="button"
              >
                Male
              </button>
              <button
                class="filter-group__filter inactive"
                type="button"
              >
                Female
              </button>
              <button
                class="filter-group__filter inactive"
                type="button"
              >
                unknown
              </button>
            </div>
            <div
              class="filter-group"
            >
              <div
                class="filter-group__label"
              >
                Origin
              </div>
              <button
                class="filter-group__filter inactive"
                type="button"
              >
                Earth
              </button>
              <button
                class="filter-group__filter inactive"
                type="button"
              >
                Mars
              </button>
              <button
                class="filter-group__filter inactive"
                type="button"
              >
                unknown
              </button>
            </div>
          </div>
        </div>
        <section
          class="col-lg-10 app__main-section"
        >
          <section
            class="row app__search-sort-section"
          >
            <div
              class="search-bar"
            >
              <input
                class="search-bar__input"
                id="filled-search"
                label="Search field"
                placeholder="Search by name"
                type="search"
              />
              <button
                class="search-bar__button"
                type="button"
              >
                Search
              </button>
            </div>
            <div
              class="sort"
            >
              <span
                class="sort__label"
              >
                Sort By ID
              </span>
              <button
                class="sort__button "
                type="button"
              >
                Asec
              </button>
              <button
                class="sort__button "
                type="button"
              >
                Desc
              </button>
            </div>
          </section>
          <div
            class="app app__message"
          >
            Loading....
          </div>
        </section>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
